components/Sign/SignForm.tsx

```diff
@@ -0,0 +1,217 @@
+import { useRouter } from "next/router";
+import { FormEvent, useState } from "react";
+import { requestSignIn, requestSignUp } from "@/lib/utils/api";
+import { useLogin } from "@/lib/utils/LoginContext";
+import SignInput from "./SignInput";
+import * as Styled from "./Sign.styled";
+import { setLocalStorage } from "@/lib/utils/localStorage";
+import {
+  validateEmailInput,
+  validatePasswordInput,
+  validatePasswordConfirmInput,
+} from "@/lib/utils/checkSign";
+
+interface Props {
+  signUp: boolean;
+  btnText: string;
+}
+
+const SignForm = ({ signUp, btnText }: Props) => {
+  const { setIsLogin, setUserEmail } = useLogin();
+  const router = useRouter();
+
+  const [emailValue, setEmailValue] = useState({
+    value: "",
+    errMsg: "",
+  });
+  const [passwordValue, setPasswordValue] = useState({
+    value: "",
+    errMsg: "",
+  });
+  const [passwordConfirmValue, setPasswordConfirmValue] = useState({
+    value: "",
+    errMsg: "",
+  });
+  const [isOverlap, setIsOverlap] = useState(true);
+
+  const checkOutSignIn = () => {
+    if (validateEmailInput(emailValue.value)) {
+      const errorMsg = validateEmailInput(emailValue.value);
+      setEmailValue((prev) => ({
+        ...prev,
+        errMsg: errorMsg,
+      }));
+    }
+    if (validatePasswordInput(passwordValue.value)) {
+      const errorMsg = validatePasswordInput(passwordValue.value);
+      setPasswordValue((prev) => ({
+        ...prev,
+        errMsg: errorMsg,
+      }));
+    }
+  };
+
+  const checkOutSignUp = () => {
+    if (validatePasswordInput(passwordValue.value)) {
+      const errorMsg = validatePasswordInput(passwordValue.value);
```

Comment:
같은 함수를 2번 호출하고 있는데 뭔가 좋은 방법이 없을까요..흐음
저라면 순서를 바꿔서
```suggestion
const errorMsg = validatePasswordInput(passwordValue.value);
if(errorMsg) //...
```
이런식으로 했을 것 같아요!